src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -134,7 +134,7 @@ protected static Object invokeMethod(Method thisMethod, Object instance, Object[
       }
     }
     throw new TestNGException("Data Provider " + dataProvider + " must return"
-          + " either Object[][] or Iterator<Object>[], not " + dataProvider.getReturnType());
+          + " either Object[][] or Object[] or Iterator<Object>[], not " + dataProvider.getReturnType());

```

Comment:
Could you fix the message? `Iterator<Object>[]` -> `Iterator<Object[]>`.
And fyi, `Iterator<Object>` is supported too.